WordPress/Classes/Users/UserProvider.swift

```diff
@@ -11,18 +11,40 @@ public enum UserDataStoreQuery: Equatable {
 }
 
 public protocol UserServiceProtocol: Actor {
-    var dataStore: any UserDataStore { get }
-
     func fetchUsers() async throws
 
     func isCurrentUserCapableOf(_ capability: String) async -> Bool
 
     func setNewPassword(id: Int32, newPassword: String) async throws
 
     func deleteUser(id: Int32, reassigningPostsTo newUserId: Int32) async throws
+
+    func allUsers() async throws -> [DisplayUser]
+
+    func streamSearchResult(input: String) async -> AsyncStream<Result<[DisplayUser], Error>>
+
+    func streamAll() async -> AsyncStream<Result<[DisplayUser], Error>>
+}
+
+protocol UserDataStoreProvider: Actor {
+    var userDataStore: any UserDataStore { get }
+}
+
+extension UserServiceProtocol where Self: UserDataStoreProvider {
+    func allUsers() async throws -> [DisplayUser] {
```

Comment:
I bet this could be generic, but we can do that later.

Reply:
Do you mind elaborating on this?